File: app/market/tick_processor.py
import orjson

from app.cache.redis import redis_client

from app.core.logger import logger


class TickProcessor:

    CHANNEL_NAME = "market_ticks"

    @staticmethod
    async def process_tick(data: dict):

        normalized_tick = {
            "symbol": data.get("s"),
            "price": float(data.get("p")),
            "quantity": float(data.get("q")),
            "trade_time": data.get("T")
        }

        await redis_client.publish(
            TickProcessor.CHANNEL_NAME,
            orjson.dumps(normalized_tick).decode()
        )

        await redis_client.set(
            f"latest_price:{normalized_tick['symbol']}",
            normalized_tick["price"]
        )

        logger.info(
            f"Processed tick: {normalized_tick}"
        )

File: backend/src/trading/candles/tickProcessor.ts
import redisClient from "../../core/redis";
import { logger } from "../../core/logger";

export class TickProcessor {
  static CHANNEL_NAME = "market_ticks";

  static async processTick(data: any) {
    const normalizedTick = {
      symbol: data.s,
      price: parseFloat(data.p),
      quantity: parseFloat(data.q),
      trade_time: data.T,
    };

    await redisClient.publish(this.CHANNEL_NAME, JSON.stringify(normalizedTick));
    await redisClient.set(`latest_price:${normalizedTick.symbol}`, String(normalizedTick.price));

    logger.info(`Processed tick: ${JSON.stringify(normalizedTick)}`);
  }
}
export default TickProcessor;
